InnoGotchi.API/Infrastructure.Services/PetConditionService.cs

```diff
@@ -0,0 +1,245 @@
+                    HungerState = pet.HungerLevel
+                };
+                await _repositoryManager.FeedingHistoryRepository.CreateRecordAsync(feeding);
+
+                if (pet.HungerLevel == HungerLevel.Dead)
+                {
+                    pet.DeathDay = feeding.ChangesDate;
+                    break;
+                }
+            }
+        }
+
+        private async Task UpdateThirstyStateAsync(Pet pet)
+        {
+            var lastThirstyUpdate = pet.ThirstyStateChangesHistory.OrderBy(d => d.ChangesDate)
+                .Last();
+            var thirstyDuration = (DateTime.Now - lastThirstyUpdate.ChangesDate).TotalHours;
+            var drinking = new ThirstyStateChanges
+            {
+                ChangesDate = lastThirstyUpdate.ChangesDate
+            };
+
+            while (thirstyDuration >= PetConfiguration.DrinkingFrequencyInHours)
+            {
+                thirstyDuration -= PetConfiguration.DrinkingFrequencyInHours;
+                pet.ThirstyLevel -= 1;
+
+                drinking = new ThirstyStateChanges()
+                {
+                    PetId = pet.Id,
+                    ChangesDate = drinking.ChangesDate.AddHours(PetConfiguration.DrinkingFrequencyInHours),
+                    ThirstyState = pet.ThirstyLevel
+                };
+                await _repositoryManager.DrinkingHistoryRepository.CreateRecordAsync(drinking);
+
+                if (pet.ThirstyLevel == ThirstyLevel.Dead)
+                {
+                    if (pet.DeathDay > drinking.ChangesDate || pet.DeathDay == null)
+                        pet.DeathDay = drinking.ChangesDate;
+                    break;
+                }
+            }
+        }
+
+        public int CalculateAge(Pet pet) =>
+            pet.DeathDay != null ?
+                (pet.DeathDay - pet.Birthday).Value.Days / PetConfiguration.OnePetAgeInDays :
+                (DateTime.Now - pet.Birthday).Days / PetConfiguration.OnePetAgeInDays;
+
+        public async Task<double> CalculateHappynessDayCount(Guid petId)
+        {
+            var hungryRecords =
+                (await _repositoryManager.FeedingHistoryRepository.GetHistoryByPetIdAsync(petId, false)).ToList();
+            var thirstyRecords =
+                (await _repositoryManager.DrinkingHistoryRepository.GetHistoryByPetIdAsync(petId, false)).ToList();
+            hungryRecords.Add(new HungryStateChanges { PetId = petId, ChangesDate = DateTime.Now });
+            thirstyRecords.Add(new ThirstyStateChanges { PetId = petId, ChangesDate = DateTime.Now });
+
+            DateTime startH = DateTime.Now, endH = DateTime.Now;
+            DateTime startT = DateTime.Now, endT = DateTime.Now;
+
+            int i = 0, j = 0;
+
+            var count = 0.0;
+
+            bool emptyH = true, emptyT = true;
+            bool nextT = true, nextH = true;
+
+            while (true)
```

Comment:
Somehow when feeding the first pet I was able to get into the infinite loop. Could you check this method one more time please?

Reply:
fixed